### ISSUE TITLE
HTTPRoute not creating Lattice Service or Target Group

### ISSUE THREAD

**Issue by afroman-lw** [aws/aws-application-networking-k8s#546]
I am following the quick start guide linked below. When I get to the step to provision the deployments, services, and HTTPRoutes, the example manifests get provisioned without error. I can see the all the provisioned resources with `kubectl`, but the Lattice Service and Target Groups are not getting created in the AWS console.

I'll note that I am creating the Lattice Service Network and VPC association outside of the EKS cluster. I have also gotten the quick start guide to work with v0.0.18, but I can't get it to work with v1.0.0 or v1.0.1

https://github.com/aws/aws-application-networking-k8s/blob/main/docs/guides/getstarted.md

**Comment by mikhail-aws** [comment]
thanks for reporting, we will try to reproduce and come back to you

**Comment by solmonk** [comment]
Could you share more information about your situation, such as logs and manifests?

The biggest breaking change between v0.0.18 and v1.0.x is that the controller does not automatically create service network anymore - you will need to create a VPC Lattice service network with the same name as Gateway. (and associate cluster VPC to it)

**Comment by afroman-lw** [comment]
Sure, I manually created the Service Network named "eks-lattice-network" and manually associated it with VPC that the EKS cluster is in before deploying the manifests to the cluster.

Attached are the manifests I deployed to the cluster. What logs would be helpful and where can I find them? I apologize as I'm a K8s novice.
[manifests.zip](https://github.com/aws/aws-application-networking-k8s/files/13494082/manifests.zip)

**Comment by afroman-lw** [comment]
Still troubleshooting this. Here are the HTTPRoute events. I'm not sure what reconciling means
```
Events:
  Type    Reason           Age                  From       Message
  ----    ------           ----                 ----       -------
  Normal  Retry-Reconcile  11m (x12 over 14m)   httproute  retry reconcile...
  Normal  Reconcile        4m1s (x35 over 14m)  httproute  Adding/Updating Reconcile
```

**Comment by afroman-lw** [comment]
My Lattice network and services are in the same account. But I didn't realize there were gateway controller pods. I'm seeing this error in the pod logs
```
I1207 15:02:43.277002       1 leaderelection.go:250] attempting to acquire leader lease aws-application-networking-system/amazon-vpc-lattice.io...
E1207 15:02:43.371128       1 leaderelection.go:336] error initially creating leader election record: leases.coordination.k8s.io "amazon-vpc-lattice.io" already exists
```

**Comment by afroman-lw** [comment]
Dived deeper into the logs. This sequence of messages repeats several times.
```{"level":"info","ts":"2023-12-07T17:11:58.506Z","logger":"controller.route","caller":"controllers/route_controller.go:152","msg":"reconcile","name":"rates"}
{"level":"info","ts":"2023-12-07T17:11:58.506Z","logger":"controller.route","caller":"controllers/route_controller.go:265","msg":"Found aws-vpc-lattice for Route for rates, default"}
{"level":"info","ts":"2023-12-07T17:11:58.506Z","logger":"controller.route","caller":"controllers/route_controller.go:311","msg":"reconcile, adding or updating","name":"rates"}
{"level":"info","ts":"2023-12-07T17:11:58.506Z","logger":"controller.route","caller":"gateway/model_build_lattice_service.go:125","msg":"No custom-domain-name for route rates-default"}
{"level":"info","ts":"2023-12-07T17:11:58.609Z","logger":"controller.route","caller":"controllers/route_controller.go:155","msg":"reconcile error","name":"rates","message":"error during tg synthesis error during target group synthesis, will retry"}
```

**Comment by mikhail-aws** [comment]
@afroman-lw I will try to reproduce it and come back to you.

It's normal to have 2 controller pods, it's redundancy, they use leader election algorithm, so only one runs reconciliations and another one is stand-by.

**Comment by mikhail-aws** [comment]
@prys might be related to permissions and RAM. Can you try to use aws cli command to list service network in same account and permissions as controller? If you dont see SN in your list it's not on controller side but related to RAM and Lattice

` aws vpc-lattice list-service-networks`

you can also enable debug log for controller `log.level=debug,` https://www.gateway-api-controller.eks.aws.dev/guides/deploy/#controller-installation

after that you should see logs with "ListServiceNetworks" keyword, that will tell what controller see from the lattice API call response

**Comment by prys** [comment]
Many thanks for your response @mikhail-aws. I will raise a separate issue, because my symptoms are different to @afroman-lw and I don't want to distract from his issue.

In summary though, when working cross-account, the controller is creating the lattice service and target group when deploying the gateway api httproute resource, but is failing to associate the service with the service network.

**Comment by mrjustind** [comment]
> Many thanks for your response @mikhail-aws. I will raise a separate issue, because my symptoms are different to @afroman-lw and I don't want to distract from his issue.
> 
> 
> 
> In summary though, when working cross-account, the controller is creating the lattice service and target group when deploying the gateway api httproute resource, but is failing to associate the service with the service network.

Can you verify the service network has been shared with the remote account through resource access manager? This needs to happen outside of the controller today and can be done directly with AWS Resource Access Manager (RAM) in the console, cli, or sdk.

**Comment by afroman-lw** [comment]
Looks like it was an error in the documentation that was fixed in v1.0.2. The tag permissions in `recommended-inline-policy.json` was updated from `"tags:GetResources"` to `"tag:GetResources"`. Once I updated the policy, the HTTPRoute created the Lattice service, target groups, and associated it with the Service Network.

**Comment by mikhail-aws** [comment]
Glad it resolved, sorry I couldn't find time to debug it. Btw, you should see errors calling lattice with debug log, tag:GetResources included.
